paddle/fluid/framework/ir/fusion_group/elementwise_group_detector.cc

```diff
@@ -49,114 +54,71 @@ static bool IsSpecifiedOp(const std::unordered_set<std::string>& op_types,
   return false;
 }
 
-static bool IsBinaryOp(Node* n) {
-  if (IsSpecifiedOp(GetBinaryOpTypes(), n) && n->inputs.size() == 2U) {
-    auto* x = n->inputs[0];
-    auto* y = n->inputs[1];
+static bool IsGradOp(const Node* n) {
+  PADDLE_ENFORCE_EQ(n && n->IsOp() && n->Op(), true,
+                    platform::errors::InvalidArgument(
+                        "Expected node %p to be an operator node.", n));
+  std::string suffix = "_grad";
+  std::string op_type = n->Op()->Type();
+  size_t pos = op_type.rfind(suffix);
+  return pos != std::string::npos &&
+         pos == (op_type.length() - suffix.length());
+}
 
-    std::vector<int64_t> x_shape;
-    std::vector<int64_t> y_shape;
-    if (x && x->IsVar() && x->Var()) {
-      x_shape = x->Var()->GetShape();
-    }
-    if (y && y->IsVar() && y->Var()) {
-      y_shape = y->Var()->GetShape();
-    }
-    if (x_shape.size() == 0U || x_shape.size() != y_shape.size()) {
+static bool IsEqual(const std::vector<int64_t>& l,
```

Comment:
1. Are you implementing "equality" for std::vector<int64_t>? I think "l == r" may work because std::vector has "=="

2. If l.size() == 0 or r.size() == 0, also returns "false". So if you need this, I suggest to rename function as:

EqualAndNotEmtpy

Reply:
> Are you implementing "equality" for std::vector<int64_t>? I think "l == r" may work because std::vector has "=="

Thanks a lot to this. It will help me a lot.

> If l.size() == 0 or r.size() == 0, also returns "false". So if you need this, I suggest to rename function as: EqualAndNotEmtpy

Yes. I need them not empty. I changed the name to `IsEqualAndNotEmpty`.